source/Ship.cpp

```diff
@@ -1559,7 +1559,7 @@ void Ship::DoGeneration()
 	double maxHull = attributes.Get("hull");
 	hull = min(hull, maxHull);
 	
-	isDisabled = isOverheated || hull < MinimumHull() || (!crew && RequiredCrew());
+	isDisabled = isOverheated || (abs(hull - MinimumHull()) >= 1 && hull < MinimumHull()) || (!crew && RequiredCrew());
```

Comment:
Compensate for decimal precision errors when calculating ship disabled state by checking if the difference between the hull and minimum hull is at least 1.